internal/extsvc/github/common.go

```diff
@@ -1506,12 +1506,20 @@ func newHttpResponseState(statusCode int, headers http.Header) *httpResponseStat
 	}
 }
 
-func doRequest(ctx context.Context, oauthContext *oauthutil.OAuthContext, logger log.Logger, apiURL *url.URL, auth auth.Authenticator, rateLimitMonitor *ratelimit.Monitor, httpClient httpcli.Doer, bearerToken *auth.OAuthBearerToken, tokenRefresher oauthutil.TokenRefresher, req *http.Request, result any) (responseState *httpResponseState, err error) {
+func doRequest(ctx context.Context, logger log.Logger, apiURL *url.URL, auther auth.Authenticator, rateLimitMonitor *ratelimit.Monitor, httpClient httpcli.Doer, req *http.Request, result any) (responseState *httpResponseState, err error) {
 	req.URL.Path = path.Join(apiURL.Path, req.URL.Path)
 	req.URL = apiURL.ResolveReference(req.URL)
 	req.Header.Set("Content-Type", "application/json; charset=utf-8")
-	if auth != nil {
-		if err := auth.Authenticate(req); err != nil {
+
+	var autherWithRefresh auth.AuthenticatorWithRefresh
+	if auther != nil {
+		autherWithRefresh, ok := auther.(auth.AuthenticatorWithRefresh)
```

Comment:
(I know this branch is not supposed to work/compile, but couldn't resist:)

You're shadowing `autherWithRefresh` instead of setting the `var autherWithRefresh auth.AuthenticatorWithRefresh`.

```suggestion
        var ok bool
		autherWithRefresh, ok = auther.(auth.AuthenticatorWithRefresh)
```

Reply:
Thanks, this could have been actual hell to debug